lib/src/mac/mod.rs

```diff
@@ -37,10 +37,11 @@ macro_rules! get_cfg {
 /// ```
 #[macro_export]
 macro_rules! sql {
-	($($query:tt)*) => {
-		match $crate::sql::parse(stringify!($($query)*)) {
+	($($query:tt)*) => {{
+		let query = format!("{}", stringify!($($query)*)).replace(" :: ", "::");
```

Comment:
I don't think it's going to be this simple to fix this issue. The problem with this is that it replaces any string that contains "  :: " even when that string is not part of the syntax.

Reply:
A quick and simple but dirty workaround for this is probably to just duplicate the tags in https://github.com/surrealdb/surrealdb/blob/main/lib/src/sql/function.rs that contain `::` to also include the variants that contain spaces. For example change `tag("time::now"),` to `tag("time::now"), tag("time :: now"),`. Stripping the whitespace beforehand in those `function_*` functions probably also works but that brings an allocation with it.

For now I think it's better to just avoid the `sql` macro for queries that contain functions. I'm working on a compile time `sql` macro. It will include a proper fix for this.

Reply:
@rushmorem Please share with me your progress on the compile time `sql` macro. Would be interested to see it working. Thanks

Reply:
@rushmorem SQL-prepared statement functionality as it is in sqlx or diesel would be probably useful too ...